src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java

```diff
@@ -121,24 +125,54 @@ protected boolean shootPlayer(Player p, Block block, PortalType type) {
         return playerBounced;
     }
 
+    @Nullable
+    private Location getLocation(Entity e, Location currentLocation, PortalType type, String currentWorld, String linkedWorld) {
+        Location newTo;
+        if (currentWorld.equalsIgnoreCase(linkedWorld)) {
+            newTo = null;
+        } else if (linkedWorld != null) {
+            newTo = this.linkChecker.findNewTeleportLocation(currentLocation, linkedWorld, e);
+        } else if (this.nameChecker.isValidNetherName(currentWorld)) {
+            if (type == PortalType.NETHER) {
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getNormalName(currentWorld, PortalType.NETHER), e);
+            } else {
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getEndName(this.nameChecker.getNormalName(currentWorld, PortalType.NETHER)), e);
+            }
+        } else if (this.nameChecker.isValidEndName(currentWorld)) {
+            if (type == PortalType.NETHER) {
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getNetherName(this.nameChecker.getNormalName(currentWorld, PortalType.ENDER)), e);
+            } else {
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getNormalName(currentWorld, PortalType.ENDER), e);
+            }
+        } else {
+            if (type == PortalType.ENDER) {
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getEndName(currentWorld), e);
+            } else {
+                newTo = this.linkChecker.findNewTeleportLocation(currentLocation, this.nameChecker.getNetherName(currentWorld), e);
+            }
+        }
+
+        return newTo;
+    }
+
```

Comment:
The cognitive complexity is quite high for this method. It could be a good place to introduce a new class to handle this.